client/src/components/molecules/TaskList/index.tsx

```diff
@@ -127,19 +130,10 @@ const TaskList: React.FC<Props> = (props): JSX.Element => {
     setIsTaskCompleted(!isTaskCompleted)
     useHandleCompleteTask(task?.id)
   }
-  const onSearchChange = (e: any): void => {
-    const value = e.target.value
-    if (value.length === 0) {
-      filterMembersName(parseInt(id as string), value)
-    }
-  }
-  const onSearch = (e: any): void => {
-    const value = e.target.value
-    const isEnter = e.key === 'Enter' || e.keyCode === 13
 
-    if (!isEnter) return
-    filterMembersName(parseInt(id as string), value)
-  }
+  useEffect(() => {
+    filterMembersName(parseInt(id as string), search)
+  }, [debouncedSearch])
```

Comment:
What is the default type of `id`? If it was previously an integer then no need to parse.

Reply:
this is from query , its a string